wordpress-mobile/WordPress-iOS
# Reader: Add "Your Tags" feed menu item

Fixes #22954 

## Description

- Add the "Your Tags" feed (✅ feature flag)
- Separate menu items into sections (❌  feature flag)
- Update "Lists" feed icon (❌  feature flag)

## Screenshots

| Before | After |
|---|---|
| ![Screenshot 2024-04-09 at 6 02 20 PM](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/5ec38ef3-3fa0-4eb5-afb5-f6652e39de14) | ![Screenshot 2024-04-09 at 5 57 54 PM](https://github.com/wordpress-mobile/WordPress-iOS/assets/2454408/5e9c1308-9492-4925-a218-a4e1b9f7f32b) |

## Testing

To test:
- Use an account which has > 2 lists
- Launch Jetpack and login
- Enable the "Reader Tags Feed" feature flag (may require restart)
- Navigate to Reader
- Tap on the navigation menu button
- 🔎 **Verify** there are dividers:
  - Before the "Saved" feed
  - Before the "Your Tags" feed
- 🔎 **Verify** the "Lists" feed has an icon and matches the design
- Tap on the "Your Tags" feed (Note: the feed doesn't do anything at this point, that will be a future PR)
- 🔎 **Verify** `🔵 Tracked: reader_dropdown_menu_item_tapped <id: tags>` is sent

## Regression Notes
1. Potential unintended areas of impact
Should be none.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


Testing checklist:
